### DISCUSSION TITLE
質問・要望

### DISCUSSION THREAD

**Discussion by yuru7** [yuru7/HackGen#32] [General]
質問や要望をフランクに投稿するためのスペースです。

**Comment by NONONOexe** [comment]
HackGenみやすくていろんなとこで利用させていただいてます．
現在chocolateyに対応していますが，最近自分はwingetでソフトを管理することが多いのですが，HackGenもwingetでインストールできるようになったらうれしいです．

**Reply by yuru7** [reply]
私はパッケージ管理まで手を広げることは考えていないので、有志の方任せにするつもりです。chocolateyも有志の方が登録してくださっているので、どう管理がされているかを私は知りません。(本リポジトリと連動してバージョンが追随するようにはなっているようですが)
どこかでwingetはフォントインストールに未対応というのを見かけた気がしますが、出来るようになったら手軽でいいですね。

**Reply by dfirr** [reply]
3年以上前の話題ですが、私もwingetでインストールできたらいいなと思ったので、パッケージ化して提出し、無事にマージされました。`winget install hackgen`でインストール可能です。

**Reply by NONONOexe** [reply]
大変助かります！

**Comment by mondy** [comment]
HackGen NFについてConsoleのみの提供となっていますが、隣の文字と詰まってしまい見づらいです。
実際に使ってる人たちはどのように使っているのか知りたくて、質問させていただきました。

久々にフォントをバージョンアップしようとしたら、NerdFontを全角として扱っているバージョンが無くなっていたので驚きました。
私の環境はWindows10で、Alacritty(とStarship)でターミナルを表示するためにこのフォントを使用しています。
Alacrittyではoffsetとglyph_offsetという設定項目があり、文字間隔を調整できますが、すべての文字に適用されるので、NerdFontに合わせると半角英数などが不格好になります。

実際にこのフォントを使っている方はどういう感じで使っているのでしょうか。良ければ教えて頂きたいです。
例えば、以下のような感じでしょうか。
- 自力でFontForgeなどを使ってフォント自体を調整している
- 他のターミナルソフトや環境だと個別の文字ごとに調整する項目があって特に問題となってない
- 左右に半角空白を挟んだり、詰まっていても特に気にならない

Consoleのみの提供となっているのは、メンテナンスの問題でしょうか。
もし可能であればどういう経緯で無くなったのか知りたいです。

**Reply by yuru7** [reply]
> Consoleのみの提供となっているのは、メンテナンスの問題でしょうか。

- 問題にならないと思える範囲でバリエーションを絞ることでメンテナンス負担を減らしたかった
- NerdFonts本家でも標準のバリエーションでは、設定されている幅よりグリフの方が大きい
  - 下図はNerdFonts本家リポジトリでリリースされているHackのNerdFonts版。X幅が1233に対してグリフの横幅が飛び出している
  - 右に半角スペースが入ることが想定されていると思われる

![image](https://github.com/yuru7/HackGen/assets/13458509/6b44f348-caad-4275-86fd-3c56b8422543)

私自身、あまりNerdFontsのヘビーユーザーではないのですが、使っているツール郡の範囲ではアイコングリフの右に半角スペースが入っているため表示上の不具合は起きていないんですよね。

**Reply by mondy** [reply]
ご回答いただきありがとうございます。

> 右に半角スペースが入ることが想定されていると思われる

アイコンの右側にスペースを入れるようにプロンプトを考えようと思いましたが、納得できるデザインにならなかったので諦めました。

HackGen Console NFを直接使う方法ではないですが、WezTermというターミナルエミュレーターを使うことで、一応解決したので共有します。
WezTermにはNerdFontが組み込まれており、それを使うことで全角幅でアイコングリフが表示できました。

**Comment by superuser7777** [comment]
Hackgen全端末に入れて快適に愛用しておりますm(_ _)m
よく使うファイラーのbrootがvscode iconに対応しているとのことで、もしvscode iconも取り込めたら嬉しいです。
